system/jlib/jsmartsock.cpp

```diff
@@ -428,6 +428,34 @@ void CSmartSocketFactory::setStatus(SocketEndpoint &ep, bool status)
         ss->status=status;
 }
 
+StringBuffer & CSmartSocketFactory::getUrlStr(StringBuffer &url, bool useHostName, unsigned dnsIntervalSecs)
+{
+	SmartSocketEndpoint * sep = nextSmartEndpoint();
+	if (sep)
+	{
+		SocketEndpoint ep;
+
+		if(useHostName)
+		{
+			ep = sep->ep;
```

Comment:
Because we're constructing the URL based on the hostname, but perhaps the else portion does require a checkhost()